Kudu.Core/Jobs/JobsFileWatcher.cs

```diff
@@ -108,7 +111,7 @@ private void StartWatcher(object state)
                 lock (_lockObject)
                 {
                     // Check if there is a directory we can listen on
-                    if (!FileSystemHelpers.DirectoryExists(_watchedDirectoryPath))
+                    if (!FileSystemHelpers.DirectoryExists(_jobDirectoryPath))
                     {
                         // If not check again in 30 seconds
                         _startFileWatcherTimer.Change(CheckForWatcherTimeout, Timeout.Infinite);
```

Comment:
Unrelated question: the comment says 30 seconds, but the timeout is Timeout.Infinite.  And you're doing the wait while holding _lockObject.  Doesn't this seem dangerous?

Reply:
The 2nd argument represents [period](https://msdn.microsoft.com/en-us/library/317hx6fa(v=vs.110).aspx) of the timer.    Infinite meant no period (just do one time timer per due time).   Make sense?

Reply:
Ah, so the 30-second timeout is wired into the _startFileWatcherTimer object.  Got it.  Thx!